src/thinkspace/client/thinkspace-builder-rat/addon/steps/settings.coffee

```diff
@@ -18,73 +20,152 @@ export default step.extend
+  trat_phase:       ember.computed.reads 'manager.trat_phase'
 
-  initialize: ->
-    model = @get('builder.model')
-    @set 'model', model
-    @query_assessments().then (assessments) =>
-      @set('assessments', assessments)
-      @init_assessments()
-      @query_phases().then =>
-        @create_changeset()
-        @set_all_data_loaded()
-
-  query_phases: ->
-    new ember.RSVP.Promise (resolve, reject) =>
-      irat_assessment = @get('irat_assessment')
-      trat_assessment = @get('trat_assessment')
+  irat_assessment:  ember.computed.reads 'manager.irat'
+  trat_assessment:  ember.computed.reads 'manager.trat'
 
-      promises =
-        irat: irat_assessment.get('authable')
-        trat: trat_assessment.get('authable')
+  sync_assessments: ember.computed.reads 'model.sync_rat_assessments'
+  is_ifat:          ember.computed.reads 'trat_assessment.settings.questions.ifat'
+  is_req_just:      ember.computed.reads 'irat_assessment.settings.questions.justification'
 
-      ember.RSVP.hash(promises).then (results) =>
-        @set('irat_phase', results.irat)
-        @set('trat_phase', results.trat)
-        resolve()
+  create_changesets: ->
+    model           = @get('model')
+    irat_assessment = @get('irat_assessment')
+    trat_assessment = @get('trat_assessment')
 
-  query_assessments: ->
-    new ember.RSVP.Promise (resolve, reject) =>
-      model = @get('model')
+    v_integer  = totem_changeset.vnumber({integer: true})
+    v_positive = totem_changeset.vnumber({positive: true})
+    v_presence = totem_changeset.vpresence(true)
+
+    changeset      = totem_changeset.create(model)
+    irat_changeset = totem_changeset.create(irat_assessment)
+    trat_changeset = totem_changeset.create(trat_assessment)
 
-      query =
-        id: model.get('id')
-        componentable_type: ns.to_p('ra:assessment')
-      options =
-        action: 'phase_componentables'
-        model: ns.to_p('ra:assessment')
+    scoring_changeset = totem_changeset.create(irat_assessment.get('settings.scoring'),
+      correct:           [v_integer, v_presence, v_positive],
+      no_answer:         [v_integer, v_presence],
+      #attempted:         [v_integer, v_presence, v_positive],
+      #incorrect_attempt: [v_integer, v_presence, v_positive]
+    )
 
-      tc.query_action(ns.to_p('assignment'), query, options).then (assessments) =>
-        resolve(assessments)
-      , (error) => reject error
+    trat_scoring_cs = totem_changeset.create(trat_assessment.get('settings.scoring'),
+      attempted:         [v_integer, v_presence, v_positive],
+      incorrect_attempt: [v_integer, v_presence, v_positive]
+    )
 
-  init_assessments: ->
-    assessments = @get('assessments')
-    manager     = @get('manager')
+    @init_incorrect_attempt(trat_scoring_cs)
 
-    irat = assessments.findBy 'is_irat', true
-    trat = assessments.findBy 'is_trat', true
+    changeset.set('show_errors', true)
```

Comment:
There is already a function for this, something like `show_errors_on()` - look at the changeset file.